plugins/rapid7_insightvm/komand_rapid7_insightvm/triggers/scan_completion/trigger.py

```diff
@@ -0,0 +1,267 @@
+
+        # Initialize trigger cache at startup
+        self.logger.info("Initialising trigger cache...")
+        util.write_to_cache(self.CACHE_FILE_NAME, json.dumps(latest_scan_id))
+
+        while True:
+            # Open cache
+            starting_point = Cache.get_cache_site_scans()
+
+            latest_scan_id = self.find_latest_completed_scan(site_id, cached=True)
+
+            # Check if latest is in cache
+            if latest_scan_id == starting_point:
+                self.logger.info("No new scans, sleeping 1 minute.")
+                time.sleep(60)
+                continue
+
+            results = self.get_results_from_latest_scan(params=params, scan_id=int(latest_scan_id))
+
+            # Submit scan for trigger
+            for item in results:
+                self.send(
+                    {
+                        Output.ASSET_ID: item.get("asset_id"),
+                        Output.IP: item.get("ip_address"),
+                        Output.HOSTNAME: item.get("hostname"),
+                        Output.VULNERABILITY_INFO: item.get("vulnerability_info"),
+                    }
+                )
+
+            # Update cache
+            try:
+                util.write_to_cache(self.CACHE_FILE_NAME, json.dumps(latest_scan_id))
+            except TypeError as error:
+                raise PluginException(
+                    cause="Failed to save cache to file", assistance=f"Exception returned was {error}"
+                )
+
+            # Sleep configured in minutes
+            time.sleep(params.get(Input.INTERVAL, 5) * 60)
+
+    def get_results_from_latest_scan(self, params: dict, scan_id: int) -> List[Dict[str, Union[str, int]]]:
+        """
+        Take a scan id and run a sql query to retrieve the
+        information needed for the trigger output
+
+        :param params: All of the user input params
+        :param scan_id: The ID of the scan
+
+        :return: A list of condensed and filter results for output in the trigger.
+        """
+
+        # Generate a unique name for the report
+        identifier = uuid.uuid4()
+
+        # Verify the scan_id input
+        Util.verify_scan_id_input(scan_id)
+
+        # Generate the payload for the reporting API call.
+        report_payload = {
+            "name": f"Rapid7-InsightConnect-ScanCompletion-{identifier}",
+            "format": "sql-query",
+            "query": ScanQueries.query_results_from_latest_scan(scan_id),
+            "version": "2.3.0",
+        }
+
+        self.logger.info("Getting report...")
+        report_contents = util.adhoc_sql_report(self.connection, self.logger, report_payload)
+        try:
+            csv_report = csv.DictReader(io.StringIO(report_contents["raw"]))
+        except Exception as error:
+            raise PluginException(
+                cause="Error: Failed to process query response while fetching site scans.",
+                assistance=f"Exception returned was {error}",
+            )
```

Comment:
if this gets raised here the whole trigger will stop working, is that intended as it will require the user to manually re-run the job again if there is an error here